src/resources/views/crud/fields/relationship/fetch_or_create.blade.php

```diff
@@ -0,0 +1,558 @@
+    var $ajax = element.attr('data-field-ajax') == 'true' ? true : false;
+    var $force_select = (element.attr('data-force-select') == 'true') ? true : false;
+
+    $modal.modal({ backdrop: 'static', keyboard: false, focus: false });
+
+
+    initializeFieldsWithJavascript($form);
+
+    //when you hit save on modal save button.
+    $modalSaveButton.on('click', function () {
+
+        $form = document.getElementById($fieldName+"-inline-create-form");
+
+        //this is needed otherwise fields like ckeditor don't post their value.
+        $($form).trigger('form-pre-serialize');
+
+        var $formData = new FormData($form);
+
+        //we change button state so users know something is happening.
+        //we also disable it to prevent double form submition
+        var loadingText = '<i class="fa fa-circle-o-notch fa-spin"></i> loading...';
+        if ($modalSaveButton.html() !== loadingText) {
+            $modalSaveButton.data('original-text', $(this).html());
+            $modalSaveButton.html(loadingText);
+            $modalSaveButton.prop('disabled', true);
+        }
+
+
+        $.ajax({
+            url: $inlineCreateRoute,
+            data: $formData,
+            processData: false,
+            contentType: false,
+            type: 'POST',
+            success: function (result) {
+
+                $createdEntity = result.data;
+
+                if(!$force_select) {
+                    //if developer did not force the created entity to be selected we first try to
+                    //check if created is still available upon model re-search.
+                    ajaxSearch(element, result.data);
+
+                }else{
+                    selectOption(element, result.data);
+                }
+
+                $modal.modal('hide');
+                //TODO: We should create translation string for this ?
+                new Noty({
+                    type: "info",
+                    text: 'Related entry created.',
+                }).show();
+            },
+            error: function (result) {
+
+                var $errors = result.responseJSON.errors;
+
+                let message = '';
+                for (var i in $errors) {
+                    message += $errors[i] + ' \n';
+                }
+
+                new Noty({
+                    type: "error",
+                    text: '<strong>Error creating related entry.</strong><br> '+message,
+                }).show();
+
+                //revert save button back to normal
+                $modalSaveButton.prop('disabled', false);
+                $modalSaveButton.html($modalSaveButton.data('original-text'));
+            }
+        });
+    });
+
+    $modal.on('hidden.bs.modal', function (e) {
+        $modal.remove();
+    });
+
+
+    $modal.on('shown.bs.modal', function (e) {
+        $(".loading_modal_dialog").hide();
+    });
+}
+
+//function responsible for adding an option to the select
```

Comment:
## Relationship Issue 11

I've added a ```fetch_or_create``` field to MonsterCrudController, using:
```php
            $this->crud->addField([    // Relationship
                'label'     => 'Relationship (n-n with InlineCreate; Fetch using AJAX) <span class="badge badge-warning">New in 4.1</span>',
                'type'      => 'relationship',
                'name'      => 'products',
                'entity'    => 'products',
                'attribute' => 'name',
                'ajax' => true,
                'inline_create' => ['entity' => 'product'],
                'data_source' => url($this->crud->getRoute().'/fetch/product'),
                'wrapperAttributes' => ['class' => 'form-group col-md-6'],
            ]);
```

The + Add button appeared. GOOD.
The Modal displayed. GOOD.
I was able to add a new Product, and all fields worked, out of a form that's not the simplest. VERY GOOD.

But, after the modal closed, when an option has been added to the select, that option displayed this weird ```[object Object]```thing instead of the name of the entry that was just created:
![Screenshot 2020-03-12 at 15 31 21](https://user-images.githubusercontent.com/1032474/76527124-5d742500-6477-11ea-82ed-dc124bdea360.png)

Again - the entry was inline created successfully. And if I hit submit, on Edit the name displays correctly. Just not right after it's been inline created...

Reply:
AH FUCK! I know why. Because Product is a translatable entity :-) So "name" is actually a JSON with the translations for each language. We need to account for that too...

Reply:
You can see my exact code here - https://github.com/Laravel-Backpack/Demo/pull/134

Reply:
Indeed, nice catch. Haven't been there yet hehe. Just fixed.

Reply:
Confirmed!